### ISSUE TITLE
Configurable install location for liblfortran_runtime

### ISSUE THREAD

**Issue by bkmgit** [lfortran/lfortran#2578]
Currently liblfortran_runtime is installed into share/lfortran/lib https://github.com/lfortran/lfortran/blob/main/src/runtime/legacy/CMakeLists.txt#L20-L35 It may be helpful to use the configurable CMake variable [CMake_install_libdir](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) to allow one to change this appropriately for different operating systems and architectures.

**Comment by certik** [comment]
It needs to be appropriately adjusted in the source code, so that the `lfortran` driver can find it.

**Comment by junghans** [comment]
Yeah, `share` is usually only for Architecture-independent data (see FHS), so a library shouldn't be installed there.

**Comment by certik** [comment]
@junghans where should be the correct place for the runtime library? Let's use the correct path by default.

**Comment by junghans** [comment]
Every distribution and package manager does it slightly differently, but `CMAKE_INSTALL_LIBDIR` does the right thing everywhere!

**Comment by junghans** [comment]
#2608 already takes care of this!
